connectors/byoc.py

```diff
@@ -630,26 +566,38 @@ async def _sync_done(self, job, status, result, exception=None):
             await job.fail(exception, ingestion_stats=ingestion_stats)
         elif status == JobStatus.SUSPENDED:
             await job.suspend(ingestion_stats=ingestion_stats)
-            logger.info(f"Sync suspended, Job id: {job.id}")
         else:
             await job.done(ingestion_stats=ingestion_stats)
 
         job = await job.reload()
         await self.sync_done(job)
         logger.info(
             f"[{job_id}] Sync done: {doc_updated + doc_created} indexed, {doc_deleted} "
-            f" deleted. ({int(time.time() - self._start_time)} seconds)"
+            f" deleted. ({int(time.time() - start_time)} seconds)"
         )
 
     async def sync_done(self, job):
-        self.doc_source["last_sync_status"] = job.status.value
-        self.doc_source["last_sync_error"] = job.error
-        self.doc_source["error"] = job.error
-        self.status = (
-            Status.ERROR if job.status == JobStatus.ERROR else Status.CONNECTED
+        job_status = JobStatus.ERROR if job is None else job.status
+        job_error = JOB_NOT_FOUND_ERROR if job is None else job.error
```

Comment:
I'm not sure this should even happen - raise an error in console but we should not actually save anything IMO.

Reply:
It can happen, when a job was deleted during execution (e.g. via DevTools). In such a case, we should update `last_synced_*` of the connector.